eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/UnmodifiableMutableSet.java

```diff
@@ -104,6 +104,16 @@
     @Override
     public boolean equals(Object obj)
     {
+        if (obj == this)
+        {
+            return true;
+        }
+
+        if (!(obj instanceof Set))
+        {
+            return false;
+        }
+
         return this.getMutableSet().equals(obj);
```

Comment:
In wrappers, I see how checking `== this` is a nice optimization. I don't see why we'd check `instanceof Set` though. That's the second thing the delegate will do, after its own `== this` check.

Reply:
yes, you are right, it is a useless optimization here and in the other case where there is a delegate!

Reply:
well except for synchronized implementation, that would avoid going through a lock

Reply:
@motlin @donraab I fixed that in the latest push